main/input_default.cpp

```diff
@@ -876,6 +876,10 @@ void InputDefault::joy_axis(int p_device, int p_axis, const JoyAxis &p_value) {
 
 	Joypad &joy = joy_names[p_device];
 
+	if (p_axis < 0 || p_axis >= JOY_AXIS_MAX) {
```

Comment:
I've added it and commited the change, and I tried to squash the commits but I think it didn't work well :(